src/linear-progress.jsx

```diff
@@ -146,6 +163,7 @@ const LinearProgress = React.createClass({
         borderRadius: 2,
         margin: 0,
         overflow: 'hidden',
+        visibility: this.state.visible ? 'visible' : 'hidden',
```

Comment:
Have you try to render nothing when `this.state.visible` is false instead of using `visibility`?